### PR TITLE
chore: update to use Node.js 20.x

### PR DESCRIPTION
**Pull request by owenvoke**
Related to https://worksome.atlassian.net/browse/JIRA-10657

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -96,15 +96,14 @@ function generateIgnore(ig, codeOwnerPath) {
...
+        const listFilesResponse = yield client.paginate(client.rest.pulls.listFiles, {
             owner: github.context.repo.owner,
             repo: github.context.repo.repo,
             pull_number: prNumber,
         });
-        const listFilesResponse = yield client.paginate(listFilesOptions);
         const changedFiles = listFilesResponse
-            .filter((f) => includeDeleted || f.status !== 'deleted')
-            .map((f) => f.filename);
+            .filter((file) => includeDeleted || file.status !== 'removed')
```
Comment (jeremynikolic) [inline dist/index.js:105]: 🙏

File: dist/index.js [hunk 2]
```diff
@@ -2414,7 +2447,7 @@ exports.Octokit = Octokit;
 
 Object.defineProperty(exports, "__esModule", ({ value: true }));
 
-var isPlainObject = __nccwpck_require__(558);
+var isPlainObject = __nccwpck_require__(3287);
```
Comment (jeremynikolic) [inline dist/index.js:2450]: - [x] Is that a safe thing to change the port ?

**Review by jeremynikolic — APPROVED** [review]
Looks good 👍 

Got a question regarding the port change to be safe

- [x] Is that a safe thing to change the port ?

File: dist/index.js (hunk 2 above)
Comment (owenvoke) [reply dist/index.js:2450]: Is that a port? 🤔 Or were you meaning import? 🙂 I'm not sure how NCC's `require` works, but I'd assume so... 🤷🏻

It seems to run correctly locally.

File: dist/index.js (hunk 1 above)
Comment (owenvoke) [reply dist/index.js:105]: Yeah, I noticed I'd written the wrong status so it was never actually removing deleted files. 😐

File: dist/index.js (hunk 2 above)
Comment (jeremynikolic) [reply dist/index.js:2450]: Looked like a mail server port or something but maybe it isn't 😄 
All good if you could confirm it works 👍

**Review by alberthaff — APPROVED** [review]
LGTM - but it might make sense to also ask for a review from someone with more experience with these tools.

**Pull request merged**